### PR TITLE
Fix Push Notification Problems

### PR DESCRIPTION
**Pull request by PeterNerlich**
### Short description

Fix discovered bugs in the recently merged push notification features for testing

### Proposed changes

- Timezone information is not a property of PushNotifications anymore, scheduled date is now always rendered in the timezone of the currently active region
- Languages are sorted according to the language tree of the currently active region
- If the user has access to only one region, the form won't show the regions checkboxes and POST won't contain that data. Add it back so the form data can be validated and saved properly.

### Side effects

unknown

__________________________________________________

[Pull Request Review Guidelines](https://digitalfabrik.github.io/integreat-cms/pull-request-review-guide.html)

### CODE REVIEW

File: integreat_cms/cms/views/push_notifications/push_notification_form_view.py [hunk 1]
```diff
@@ -379,3 +393,27 @@ def extract_pn_details(request, push_notification):
         "all_languages": all_languages,
         "disable_edit": disable_edit,
     }
+
+
+def patch_missing_regions_field(request):
```
Comment (timoludwig) [inline integreat_cms/cms/views/push_notifications/push_notification_form_view.py:398]: I think this is the wrong way of handling this problem. I'd prefer it if the value was contained in the html form already, and only hidden from the user via css.

Comment (PeterNerlich) [reply integreat_cms/cms/views/push_notifications/push_notification_form_view.py:398]: Yes, but I figured we needed to validate it in this case anyway (whether the user has access to it). Do you have a quick suggestion for an alternative approach?

Comment (svenseeberg) [reply integreat_cms/cms/views/push_notifications/push_notification_form_view.py:398]: In my opinion your suggestion looks valid @PeterNerlich , thanks. I would rephrase the logged message a bit (could be a bug, not a manipulation):
```python
            logger.warning(
                "%r is not allowed to post news to %r (allowed regions: %r)",
                request.user,
                push_notification_instance,
                request.available_regions
            )
            raise PermissionDenied
```

**Comment by svenseeberg** [comment]
While testing I noticed that we could easily improve the error message about the not assigned regions a bit:
```
You are not allowed to edit this message. This news message is also assigned to regions you don't have access to: Nationalparklandkreis Nürnberg. Please ask the one responsible for all of these regions or contact an administrator.
```
The point being: first tell the user what is going on. Then explain why.

File: integreat_cms/cms/views/push_notifications/push_notification_form_view.py [hunk 2]
```diff
@@ -99,12 +99,12 @@ def get(self, request, *args, **kwargs):
             if details["disable_edit"]:
                 not_accessible_regions_warning = mark_safe(
                     __(
+                        _("You are not allowed to edit this message."),
                         _(
-                            "This news message is also assigned to regions you don't have access to: {}."
+                            "This news message belongts to regions you don't have access to: {}."
```
Comment (PeterNerlich) [inline integreat_cms/cms/views/push_notifications/push_notification_form_view.py:104]: Typo:
```suggestion
                            "This news message belongs to regions you don't have access to: {}."
```

File: integreat_cms/locale/de/LC_MESSAGES/django.po [hunk 3]
```diff
@@ -8672,6 +8672,23 @@ msgstr ""
 "angezeigt, haben allerdings keine Auswirkung auf die gesendete Push-"
 "Benachrichtigung."
 
+#: cms/views/push_notifications/push_notification_form_view.py
+msgid "You are not allowed to edit this message."
+msgstr "Sie dürfen diese Nachricht nicht bearbeiten."
+
+#: cms/views/push_notifications/push_notification_form_view.py
+msgid "This news message belongts to regions you don't have access to: {}."
```
Comment (PeterNerlich) [inline integreat_cms/locale/de/LC_MESSAGES/django.po:8680]:
```suggestion
msgid "This news message belongs to regions you don't have access to: {}."
```

**Comment by PeterNerlich** [comment]
Oh no, of course it's the pesky *Too many statements* again :see_no_evil:

File: integreat_cms/cms/views/push_notifications/push_notification_form_view.py [hunk 4]
```diff
@@ -263,6 +263,14 @@ def post(self, request, *args, **kwargs):
             for form in pnt_formset:
                 if not form.is_valid():
                     form.add_error_messages(request)
+        elif set(pn_form.cleaned_data["regions"]).difference(request.available_regions):
+            logger.warning(
+                "%r is not allowed to post news to %r (allowed regions: %r)",
+                request.user,
+                push_notification_instance,
+                request.available_regions
+            )
```
Comment (PeterNerlich) [inline integreat_cms/cms/views/push_notifications/push_notification_form_view.py]: This message only makes sense with `pn_form.cleaned_data["regions"]`:
```suggestion
            logger.warning(
                "%r is not allowed to post news to %r (allowed regions: %r)",
                request.user,
                pn_form.cleaned_data["regions"],
                request.available_regions,
            )
```

Comment (svenseeberg) [reply integreat_cms/cms/views/push_notifications/push_notification_form_view.py]: True, I mixed up the parameters.

**Comment by svenseeberg** [comment]
@PeterNerlich and I found an interesting issue: while the dev server was running, we sent a message via
```
integreat-cms-cli send_push_notifications
```
The database shows the message as sent, the log output was as expected:
```
integreat=# SELECT * FROM cms_pushnotification WHERE id=9;
 id | channel | draft |           sent_date           |         created_date          |      mode      |  scheduled_send_date   
----+---------+-------+-------------------------------+-------------------------------+----------------+------------------------
  9 | news    | f     | 2023-07-27 20:57:53.178401+02 | 2023-07-27 20:51:34.231419+02 | ONLY_AVAILABLE | 2023-07-27 20:55:00+02
```

However, the dev server shows the message as not being sent: http://localhost:8000/testumgebung/push-notifications/de/9/edit/ says `Message not sent yet`. Probably some weird caching bug? Restarting the dev server does not help, it seems.

**Review by svenseeberg — APPROVED** [review]
